client/src/holdings/AddHoldingDialog.tsx

```diff
@@ -0,0 +1,177 @@
+import './Dialog.css';
+
+import React, {useState} from 'react';
+import {Controller, useForm} from 'react-hook-form';
+import * as yup from 'yup';
+
+import {yupResolver} from '@hookform/resolvers/yup';
+import {
+    Button, CircularProgress, Dialog, DialogActions, DialogContent, DialogTitle, MenuItem, Select,
+    TextField
+} from '@material-ui/core';
+
+import {API} from '../api';
+import {CURRENCIES} from '../constants';
+import {CreateInstrumentData, Currency} from '../types';
+
+interface IFormInput {
+  name: string;
+  symbol: string;
+  category: string;
+  isin: string;
+  currency: Currency;
+  exchange: string;
+}
+
+type AddHoldingDialogProps = {
```

Comment:
Ah can you use interface for here actually, I'm gonna change them all at some point https://github.com/willbeaufoy/portfolio-tracker/issues/63